### PR TITLE
Auth routes

### LINKED ISSUE #20: Implement signup and signin functionality
Don't have to design them to the maximum for now, just need to create the routes for those pages and create basic front-end

### PR DESCRIPTION
**Pull request by moezAhmad**
### Description
**Implemented Authentication with Google OAuth and Credentials in auth_routes Branch**

This pull request introduces significant updates to the authentication system within our application, focusing on both Google OAuth and traditional sign-in with credentials. The changes are encapsulated in the auth_routes branch. Below are the key highlights of this update:

**Google OAuth Integration:** Integrated Google OAuth for user authentication. This approach simplifies the login process for users, leveraging their Google accounts for secure and quick access.

**Sign-In with Credentials:** Implemented the classic sign-in method where users can log in using their email and password. This ensures that users without a Google account or those preferring traditional login methods can access the application.

**Handling New Users in Google OAuth:** For new users signing up via Google OAuth, the system now prompts them to provide additional details necessary for creating a comprehensive database record. After this step, they are directed to create a waza account, ensuring that all users have complete and functional profiles.

**Waza Account Creation for Credential Sign-Ins:** In the case of users signing in with credentials, the system checks for the presence of a waza_id. If a waza_id is not found, indicating a new user, the application prompts the user to create a waza account. This process is based on the user's type and ensures that all users have associated waza accounts for complete functionality within the application.

### Related Issue
#20 

Reviewers
@Waleed-abdullah

### CODE REVIEW

**Comment by Waleed-abdullah** [comment]
Kindly resolve these build issues
image

File: src/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      },
+      async authorize(credentials, req) {
+        if (!credentials) {
+          return null;
+        }
+        const user = await prisma.users.findUnique({
+          where: { email: credentials.email },
+        });
+        console.log('======================authorize=====================');
+        console.log(user);
```
Comment (Waleed-abdullah) [inline src/pages/api/auth/[...nextauth].ts]:
```suggestion

```

File: src/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if (
+          user &&
+          user.password &&
+          bcrypt.compareSync(credentials.password, user.password)
+        ) {
+          console.log('user found password matched');
```
Comment (Waleed-abdullah) [inline src/pages/api/auth/[...nextauth].ts]:
```suggestion
 
```

File: src/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  callbacks: {
+    async signIn({ user, account, profile }) {
+      if (!account) return false;
+      user.provider = account.provider;
+      console.log('================user=====================');
+      console.log(user);
+      console.log('================account=====================');
+      console.log(account);
+      console.log('================profile=====================');
+      console.log(profile);
```
Comment (Waleed-abdullah) [inline src/pages/api/auth/[...nextauth].ts]:
```suggestion
```

File: src/pages/api/auth/[...nextauth].ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      if (account.provider === 'google') {
+        user.is_verified = (
+          profile as { email_verified: boolean }
+        ).email_verified;
+        const userInDb = await prisma.users.findUnique({
+          where: { email: user.email || '' },
+        });
+
+        console.log('================userInDb=====================');
+        console.log(userInDb);
```
Comment (Waleed-abdullah) [inline src/pages/api/auth/[...nextauth].ts]:
```suggestion
```

**Review by Waleed-abdullah — CHANGES_REQUESTED** [review]

**Comment by moezAhmad** [comment]
I have removed all console logs.

**Review by Waleed-abdullah — APPROVED** [review]

**Pull request merged**